lib/api/survey_service.dart

```diff
@@ -0,0 +1,17 @@
+import 'package:dio/dio.dart';
+import 'package:retrofit/retrofit.dart';
+
+import '../model/response/survey_response.dart';
+
+part 'survey_service.g.dart';
+
+@RestApi()
+abstract class SurveyService {
+  factory SurveyService(Dio dio, {String baseUrl}) = _SurveyService;
```

Comment:
@Wadeewee you mean generating the `_SurveyService` class and then copying and pasting its content in the `SurveyServiceImpl` class and then deleting the annotations and generated code? 🤔 The idea of using code generation using annotation is to reduce boilerplate code and speed up development.

Reply:
@AVI5HEK I mean like this 🙏 

```dart
abstract class SurveyService {
  Future<List<SurveyResponse>> getSurveys(
    int pageNumber,
    int pageSize,
  );
}

@RestApi()
abstract class SurveyServiceImpl extends SurveyService {
  factory SurveyServiceImpl(Dio dio, {String baseUrl}) = _SurveyServiceImpl;

  @override
  @GET('/api/v1/surveys?page[number]={number}&page[size]={size}')
  Future<List<SurveyResponse>> getSurveys(
    @Path('number') int pageNumber,
    @Path('size') int pageSize,
  );
}
```

Reply:
`SurveyServiceImpl` is not a concrete implementation. Nor do we create an abstract class because of MVVM architecture. But I get your idea. To create a fake service in the integration test, I've created an abstraction in PR #44 .